appIcons.js

```diff
@@ -542,6 +574,199 @@ const MyAppIcon = new Lang.Class({
             this._dots.queue_repaint();
     },
 
+    _enableBacklight: function() {
+        let pixBuf = this._getIconPixBuf();
+
+        // Fallback
+        if (pixBuf === null) {
+            this._iconContainer.set_style(
+                'border-radius: 5px;' +
+                'background-gradient-direction: vertical;' +
+                'background-gradient-start: #e0e0e0;' +
+                'background-gradient-end: darkgray;'
+            );
+            
+            this._dot.set_style(
+                'height: 0px;' +
+                'border: 0px solid white;' +
+                'background-color: gray;'
+            );
+
+            return;
+        }
+        
+        let colorPallete = this._calculateColorPalette(pixBuf);
+
+        this._iconContainer.set_style(
+            'border-radius: 5px;' +
+            'background-gradient-direction: vertical;' +
+            'background-gradient-start: ' + colorPallete.original + ';' +
+            'background-gradient-end: ' +  colorPallete.darker + ';'
+        );
+
+        this._dot.set_style(
+            'height: 0px;' +
+            'border: 0px solid ' + colorPallete.lighter  + ';' +
+            'background-color: ' + colorPallete.darker + ';'
+        );
+    },
+
+    _toggleBacklight: function () {
+        this._glossyBackground();
+        this._updateRunningStyle();
+        
+        if (this._dots)
+            this._dots.queue_repaint();
+    },
+
+    _disableBacklight: function() {
+        this._iconContainer.set_style(null);
+        this._dot.set_style(null);
+    },
+    
+    /**
+     * Try to get the pixel buffer for the current icon, if not fail gracefully     
+     */
+    _getIconPixBuf: function() {
+        let iconTexture = this.app.create_icon_texture(16);
+
+        if (themeLoader === null) {
+            let ifaceSettings = new Gio.Settings({ schema: "org.gnome.desktop.interface" });
+
+            themeLoader = new Gtk.IconTheme(),
+            themeLoader.set_custom_theme(ifaceSettings.get_string('icon-theme')); // Make sure the correct theme is loaded
+        }        
+
+        // Unable to load the icon texture, use fallback
+        if (iconTexture instanceof St.Icon === false) {
+            return null;
+        }
+
+        iconTexture = iconTexture.get_gicon();       
+
+        // Unable to load the icon texture, use fallback
+        if (iconTexture === null) {
+            return null;
+        }
+
+        if (iconTexture instanceof Gio.FileIcon) {
+            // Use GdkPixBuf to load the pixel buffer from the provided file path
+            return GdkPixbuf.Pixbuf.new_from_file(iconTexture.get_file().get_path());
+        }
+        
+        // Get the pixel buffer from the icon theme
+        return themeLoader.load_icon(iconTexture.get_names()[0], 24, 0);
```

Comment:
Small suggestion regarding the default size loaded. If we increase it from 24 to 64, there is quite an impact in the end result. This is of course just a suggestion! Cheers.

24px:
Small suggestion regarding the default size requested. If we increase it from 24 to 64, there is quite an impact in the end result.

24px:
![screenshot from 2017-06-12 12-19-00](https://user-images.githubusercontent.com/19195975/27043797-572f3570-4f69-11e7-85aa-391c52ac62fd.png)

48px:
![screenshot from 2017-06-12 12-19-40](https://user-images.githubusercontent.com/19195975/27043823-6d71a17e-4f69-11e7-9031-d8b908e89735.png)

64px:
![screenshot from 2017-06-12 12-20-05](https://user-images.githubusercontent.com/19195975/27043843-7f03cf16-4f69-11e7-9acd-8210ee5c0f2d.png)